tests/rptest/services/redpanda.py

```diff
@@ -1808,6 +1808,27 @@ def metrics(
                 pod_name).decode()
         return text_string_to_metric_families(text)
 
+    def _extract_samples(self, metrics, sample_pattern: str,
+                         pod) -> list[MetricSamples]:
+        found_sample = None
+        sample_values = []
+
+        for family in metrics:
+            for sample in family.samples:
+                if sample_pattern not in sample.name:
+                    continue
+                if not found_sample:
+                    found_sample = (family.name, sample.name)
+                if found_sample != (family.name, sample.name):
+                    raise Exception(
+                        f"More than one metric matched '{sample_pattern}'. Found {found_sample} and {(family.name, sample.name)}"
```

Comment:
If this would be used somewhere else, it _might_ be a good idea to create a custom exception and then call it

Something like:
# If more than one unique match is found, raise the exception
    if len(found_samples) > 1:
        raise MultipleMetricsMatchedException(sample_pattern, found_samples)

Reply:
good point if there is a test case in the future that needs to switch on this specific exception failure.

the `_extract_samples()` method is used by `metrics_sample()` which gets called by a lot of tests, so my approach for this PR is to keep existing behavior while adding the complimentary `metrics_*()` methods to the `RedpandaServiceCloud` class that was missing.